peak/family.py

```diff
@@ -193,7 +193,13 @@ def assemble(self, locals, globals, **kwargs):
         s_deco = super().assemble(locals, globals, **kwargs)
         def deco(cls):
             if issubclass(cls, BlackBox):
-                cls.__call__ = BlackBox.__call__
+                old_call = cls.__call__
+
+                cls.__call__ = BlackBox.create_call()
```

Comment:
make create a call a `@classmethod` then just call `cls.create_call()` which does all this in place. e.g.

```Python
@classmethod
def create_call(cls):
    def call(...): ...
    input_t = ...
    cls.__call__ = call
    cls.__call__.input_t = input_t
```
The current solution kinda has a weird intertangling of responsibilities for each function.  
At least move the setting of input_t into `create_call` so this is just `cls.__call__ = BlackBox.create_call()`

Reply:
Fixed